Sources/FluentSQL/SQLSchemaConverter.swift

```diff
@@ -83,12 +86,16 @@ public struct SQLSchemaConverter {
                     columns: local.map(self.fieldName),
                     references: reference
                 ),
-                name: SQLIdentifier("fk:\(name)")
+                name: SQLIdentifier("fk:\(normalizeSQLConstraintIdentifier(name))")
             )
         case .custom(let any):
             return custom(any)
         }
     }
+    
+    private func normalizeSQLConstraintIdentifier(_ identifier: String) -> String {
+        Data(Insecure.SHA1.hash(data: Data(identifier.utf8))).base64EncodedString()
```

Comment:
As before mentioned, Crypto is a massive dependency to pull in just for this. This also makes error messages about constraint violations utterly opaque - `insert on table 'foo' violates foreign key constraint '7d97e98f8af710c7e7fe703abc8f639e0ee507c4'` tells me _nothing_ about which part of my data is wrong. I suggest:

1) Adding a `maximumConstraintIdentifierLength` field to `SQLDialect` down in SQLKit, and until that's done, using the minimum of the maximum lengths between all of Fluent's supported drivers (between MySQL, Postgres, SQLite, and Mongo, such a minimum would be something like 56 UTF-8 bytes - note: not characters!).
2) If the generated `identifier` name doesn't exceed that limit, leave it as-is.
3) Otherwise, perform truncation by dropping `lengthExceedingLimit / 2` characters on either side of the index `identifier.count / 2`, something along the lines of `identifier.prefix(upTo: Swift.max(identifier.startIndex, (identifier.count >> 1) - ((identifier.utf8.count - limit) >> 1)) + identifier.suffix(maxLength: (identifier.count >> 1) - ((identifier.utf8.count - limit) >> 1))` (WARNING: this is _DEFINITELY_ wrong, I scribbled it down off the top of my head just now here in the GitHub comment field, it's just a very general idea of the concept).

This way not only do you not need a Crypto dependency, you preserve at least _some_ of the original identifier, hopefully enough to still tell the user what went wrong. If you find this causes collisions, you can uniquely by dropping an additional few characters from the limit and unconditionally appending the index of the identifier in the `constraints` array (this won't help as much in MySQL where constraint names have are database-global, but an atomic or locked incrementing global value would do for that).